contracts/src/Surge.sol

```diff
@@ -7,92 +7,61 @@ pragma solidity ^0.8.0;
 // █▀ █░█ █▀█ █▀▀ █▀▀   █░█░█ █▀█ █▀▄▀█ █▀▀ █▄░█
 // ▄█ █▄█ █▀▄ █▄█ ██▄   ▀▄▀▄▀ █▄█ █░▀░█ ██▄ █░▀█
 import "hardhat/console.sol";
-import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
-import "@openzeppelin/contracts/interfaces/IERC20.sol";
-import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
-import "@openzeppelin/contracts/utils/Counters.sol";
-import "@openzeppelin/contracts/utils/Address.sol";
-import "@openzeppelin/contracts/utils/Strings.sol";
-import "@openzeppelin/contracts/utils/Context.sol";
-import "@openzeppelin/contracts/utils/math/SafeMath.sol";
-
+import "erc721a/contracts/ERC721A.sol";
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import '@openzeppelin/contracts/utils/cryptography/MerkleProof.sol';
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/finance/PaymentSplitter.sol";
 
-contract Surge is ERC721, ReentrancyGuard, Ownable, ERC721Enumerable {
-    using Counters for Counters.Counter;
-    using Address for address;
-    using SafeMath for uint256;
+contract Surge is ERC721A, ReentrancyGuard, Ownable, PaymentSplitter {
     using Strings for uint256;
 
-    /*----------------------------------------------*/
-    /*                    STATE                    */
-    /*--------------------------------------------*/
-    Counters.Counter private _tokenIds;
-
-    uint8 public constant MAX_PER_USER = 5;
-    uint8 public constant MAX_RESERVED_TOKENS = 120;
-    uint8 public totalGiftMints = 0;
-    bool public saleIsActive;
-    bool public presaleIsActive;
+    bytes32 public merkleRoot;
 
-    uint16 public constant MAX_TOKENS = 2500;
-    uint256 public constant TOKEN_PRICE = 50000000000000000; //0.05ETH
+    uint256 public totalGiftMints = 0;
```

Comment:
What's the difference between this `totalGiftMints` and the `maxReserved` variable?

Reply:
I am not sure why she had this is in here, I will remove it. I think it was being used to make sure we did not mint more than the max reserved. Since we can pass in the amount with the erc721a we don't need this.